# Developed in house to disable MD4
# Internal only, will not be submitted upstream.


--- a/ext/openssl/openssl.c
+++ b/ext/openssl/openssl.c
@@ -67,7 +67,9 @@
 
 #define OPENSSL_ALGO_SHA1 	1
 #define OPENSSL_ALGO_MD5	2
+#ifdef HAVE_OPENSSL_MD4_H
 #define OPENSSL_ALGO_MD4	3
+#endif
 #ifdef HAVE_OPENSSL_MD2_H
 #define OPENSSL_ALGO_MD2	4
 #endif
@@ -1355,9 +1357,11 @@ static EVP_MD * php_openssl_get_evp_md_f
 		case OPENSSL_ALGO_MD5:
 			mdtype = (EVP_MD *) EVP_md5();
 			break;
+#ifdef HAVE_OPENSSL_MD4_H
 		case OPENSSL_ALGO_MD4:
 			mdtype = (EVP_MD *) EVP_md4();
 			break;
+#endif
 #ifdef HAVE_OPENSSL_MD2_H
 		case OPENSSL_ALGO_MD2:
 			mdtype = (EVP_MD *) EVP_md2();
@@ -1490,7 +1490,9 @@ PHP_MINIT_FUNCTION(openssl)
 	/* signature algorithm constants */
 	REGISTER_LONG_CONSTANT("OPENSSL_ALGO_SHA1", OPENSSL_ALGO_SHA1, CONST_CS|CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("OPENSSL_ALGO_MD5", OPENSSL_ALGO_MD5, CONST_CS|CONST_PERSISTENT);
+#ifdef HAVE_OPENSSL_MD4_H
 	REGISTER_LONG_CONSTANT("OPENSSL_ALGO_MD4", OPENSSL_ALGO_MD4, CONST_CS|CONST_PERSISTENT);
+#endif
 #ifdef HAVE_OPENSSL_MD2_H
 	REGISTER_LONG_CONSTANT("OPENSSL_ALGO_MD2", OPENSSL_ALGO_MD2, CONST_CS|CONST_PERSISTENT);
 #endif
